language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/InvocationNodeContextProvider.java

```diff
@@ -111,10 +114,21 @@ public void sort(BallerinaCompletionContext context, T node, List<LSCompletionIt
             } else if (parameterSymbol.isEmpty()) {
                 completionItem.getCompletionItem().setSortText(SortingUtil.genSortText(
                         SortingUtil.toRank(context, completionItem)));
+            } else if (completionItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
+                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) completionItem;
+                if (symbolCompletionItem.getSymbol().isPresent() &&
+                        symbolCompletionItem.getSymbol().get().kind() == SymbolKind.RESOURCE_METHOD) {
+                    completionItem.getCompletionItem().setSortText(
+                            SortingUtil.genSortTextByAssignability(context, completionItem, parameterSymbol.get()) +
+                                    SortingUtil.genSortText(rankOffset));
+                }
+                completionItem.getCompletionItem().setSortText(
+                        SortingUtil.genSortTextByAssignability(context, completionItem, parameterSymbol.get()));
```

Comment:
I think we can refactor this, to do this sorting inside `ClientResourceAccessActionNodeContext`. Shall we try it?

Reply:
In `ClientResourceAccessActionNodeContext`, it simply invokes the sorting method of its parent.

https://github.com/ballerina-platform/ballerina-lang/blob/bf6b50f6b0c5dca37e978d1e0cbb9b4c4f5cfad0/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ClientResourceAccessActionNodeContext.java#L214

It would be difficult to move the logic to this context.

Reply:
The variable `rankOffset` used in this super method is only used for client resource action. So I think it is better to move this into `ClientResourceAccessActionNodeContext`.